modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/tx/ClientTransactionCommitRequest.java

```diff
@@ -32,14 +33,20 @@ public class ClientTransactionCommitRequest {
      *
      * @param in        Unpacker.
      * @param resources Resources.
+     * @param metrics   Metrics.
      * @return Future.
      */
-    public static CompletableFuture<Void> process(ClientMessageUnpacker in, ClientResourceRegistry resources)
+    public static CompletableFuture<Void> process(
+            ClientMessageUnpacker in,
+            ClientResourceRegistry resources,
+            ClientHandlerMetricSource metrics)
             throws IgniteInternalCheckedException {
         long resourceId = in.unpackLong();
 
         Transaction t = resources.remove(resourceId).get(Transaction.class);
 
+        metrics.transactionsActiveDecrement();
+
         return t.commitAsync();
```

Comment:
Should not `metrics.transactionsActiveDecrement()` be in `t.commitAsync().then(...)`?

Reply:
Fixed.